src/gemm.jl

```diff
@@ -35,7 +35,7 @@ for (gemm, elt) in gemm_datatype_mappings
                                beta::$(elt), C::Ptr{$elt})
             # Convert our compile-time transpose marker to a char for BLAS
             convtrans(V::Val{false}) = 'N'
-            convtrans(V::Val{true})  = 'T'
+            convtrans(V::Val{true})  = $elt <: Complex ? 'C' : 'T'
```

Comment:
I believe that the underlying `{S,D,C,Z}GEMM` routine should correctly interpret `'C'`, even if it's a real-valued input.  In the case of real-valued input, `'C'` should be equivalent to `'T'`.

Reply:
I thought so, too, but buildkite was consistently failing on the GPU. Therefore, I implemented the branch above.